RFS/build-preloaded-source-image.gradle

```
@@ -0,0 +1,115 @@
+import com.bmuschko.gradle.docker.tasks.container.*
+import com.github.dockerjava.core.DefaultDockerClientConfig
+import com.github.dockerjava.core.DockerClientImpl
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient
+
+def createDockerClient() {
+    def config = DefaultDockerClientConfig.createDefaultConfigBuilder().build()
+    def httpClient = new ApacheDockerHttpClient.Builder()
+            .dockerHost(config.getDockerHost())
+            .sslConfig(config.getSSLConfig())
+            .maxConnections(100)
+            .build()
+
+    return DockerClientImpl.getInstance(config, httpClient)
+}
+
+def uniqueId = UUID.randomUUID();
+def myNetworkName = "rfs-preload-source-${uniqueId}";
+def createNetworkTask = task createNetwork(type: Exec) {
+    commandLine 'docker', 'network', 'create', myNetworkName
+    doLast {
+        println 'Network created'
+    }
+}
+task createInitialElasticsearchContainer(type: DockerCreateContainer) {
+    dependsOn createNetwork, buildDockerImage_emptyElasticsearchSource
+    targetImageId 'migrations/empty_elasticsearch_source:latest'
+    containerName = "elasticsearch-${uniqueId}"
+    hostConfig.network = myNetworkName
+    hostConfig.dns = ['elasticsearch']
+    networkAliases = ['elasticsearch']
+    hostName = 'elasticsearch'
+}
+
+def startSourceTask = task startInitialElasticsearchContainer(type: DockerStartContainer) {
+    dependsOn createInitialElasticsearchContainer
+    targetContainerId createInitialElasticsearchContainer.getContainerId()
+}
+
+task createClientContainer(type: DockerCreateContainer) {
+    dependsOn startInitialElasticsearchContainer, buildDockerImage_trafficGenerator
+    targetImageId 'migrations/osb_traffic_generator:latest'
+    containerName = "traffic-generator-container-${uniqueId}"
+    hostConfig.network = myNetworkName
+    //cmd = ['skip_dataset']
+    cmd = ['default_osb_test_workloads', "http://elasticsearch:9200"]
+}
+
+def startClientTask = task startClientContainer(type: DockerStartContainer) {
+    dependsOn createClientContainer
+    targetContainerId createClientContainer.getContainerId()
+}
+
+task waitClientContainer(type: DockerWaitContainer) {
+    dependsOn startClientContainer
+    targetContainerId createClientContainer.getContainerId()
+}
+
+// Task to commit the source container to create a new image that will include the loaded data
+def sourceContainerCommitTask = task commitSourceContainer() {
+    dependsOn waitClientContainer
+
+    doLast {
+        def client = createDockerClient()
+        def containerId = createInitialElasticsearchContainer.getContainerId().get()
+
+        println "Committing container ${containerId} to image my-new-primary-image:latest"
```

parameterize this so it doesn't get out of date?  Note might need slightly more refactoring
```suggestion
        println "Committing container ${containerId} to image ${image}:${version}"
```

---

removed the print altogether